store-tests

```diff
@@ -117,36 +118,39 @@ def main():
                 repo = parts[1]
 
         for retry, status in enumerate(statuses):
-            if not status.get("finished"):
-                # TODO: Store partial, store as non-finished?
-                logging.debug("Test hasn't finished, skipping")
-                continue
-
-            url = status["url"]
-            waited = status["started"] - status["posted"]
-            took = status["finished"] - status["started"]
+            posted = status["posted"]  # This must be always present, lets fails if it isn't
+            url = status.get("url")
+            started = status.get("started", posted)
+            waited = started - posted
+            took = status.get("finished", started) - started
             logging.debug("Processing {0}{1} as {2} that waited {3}s and took {4}s in {5}. {6}".format(
                 context,
                 "(retry no." + str(retry) + ")" if retry else "",
                 status["state"] + (" (" + status["description"] + ")" if status.get("description") else ""),
                 waited, took, repo, url
             ))
 
+            # URL is our primary key`, so if we don't have one, just fake something unique
+            if not url:
+                url = "-".join(["xxx", opts.revision, context, str(retry)])
+
             row = cursor.execute("SELECT id FROM TestRuns WHERE url = ?", (url, )).fetchone()
             if row is not None:
                 logging.warning("Test run %s already exists as ID %i, skipping", url, row[0])
                 continue
 
             run_id = insert_run(cursor, repo, opts.revision, context, url,
-                                status["posted"], retry, waited, took, status["state"],
+                                posted, retry, waited, took, status["state"],
                                 status.get("description"))
 
-            raw_log = url[:-5] if url.endswith(".html") else url
-            logging.debug("scanning log %s", raw_log)
+            if (status.get("url")):
```

Comment:
My first instinct here was "this should be `if url:`, but of course we don't want to scan fake URLs. A small comment may be helpful for our future selves. Or better yet, move this up as `else:` to the place that sets the fake url., and then conditionalize this on `raw_url`.

BTW, this is python, no need to wrap the condition into ().

Reply:
Right, I want to keep also checks for already existing, mostly for sanity and also don't want to duplicate insertion.
I'll add comment.